src/test/fuzz/compression_roundtrip.cpp

```diff
@@ -0,0 +1,442 @@
+                    std::optional<std::vector<unsigned char>> optional_bytes = getrandbytes(32);
+                    if (!optional_bytes.has_value()) return {};
+                    std::vector<unsigned char> script_bytes = optional_bytes.value();
+                    CScript script = CScript(script_bytes.begin(), script_bytes.end());
+                    return std::make_tuple(keypair, script, redeemScripts);
+                }
+                default:
+                    assert(false);
+            }
+   	}
+
+    	bool SignTransaction(CMutableTransaction& mtx, std::vector<secp256k1_keypair> kps, std::vector<CScript> redeemScripts) {
+    		FillableSigningProvider keystore;
+    		int kps_length = kps.size();
+    		for (int kps_index = 0; kps_index < kps_length; kps_index++) {
+    			std::vector<unsigned char> secret_key(32);
+    			if (!secp256k1_keypair_sec(secp256k1_context_static, &secret_key[0], &kps.at(kps_index))) return false;
+    			CKey key;
+    			key.Set(secret_key.begin(), secret_key.end(), true);
+    			keystore.AddKey(key);
+    			if (!key.IsValid()) return false;
+    			CKey key2;
+    			key2.Set(secret_key.begin(), secret_key.end(), false);
+    			keystore.AddKey(key2);
+    			if (!key2.IsValid()) return false;
+    		}
+
+            for (size_t rss_index = 0; rss_index < redeemScripts.size(); rss_index++) {
+                keystore.AddCScript(redeemScripts[rss_index]);
+                CScript witness_output_script{GetScriptForDestination(WitnessV0ScriptHash(redeemScripts[rss_index]))};
+                keystore.AddCScript(witness_output_script);
+            }
+    		// Fetch previous transactions (inputs):
+    		std::map<COutPoint, Coin> coins;
+    		for (const CTxIn& txin : mtx.vin) {
+    			coins[txin.prevout]; // Create empty map entry keyed by prevout.
+    		}
+    		FindCoins(m_node, coins);
+
+    		std::map<int, bilingual_str> input_errors;
+    		::SignTransaction(mtx, &keystore, coins, SIGHASH_ALL, input_errors);
+    		return input_errors.size() == 0;
+    	}
+    };
+    secp256k1_context* ctx = nullptr;
+    SecpContext secp_context = SecpContext();
+    CompressionRoundtripFuzzTestingSetup* rpc = nullptr;
+    std::vector<std::tuple<uint64_t, uint64_t>> savings;
+    std::vector<std::tuple<uint256, secp256k1_keypair, std::vector<CScript>, uint32_t, int, bool>> unspent_transactions;
+    std::vector<std::tuple<uint256, secp256k1_keypair, std::vector<CScript>, uint32_t, int, CScript>> coinbase_transactions;
+    CompressionRoundtripFuzzTestingSetup* InitializeCompressionRoundtripFuzzTestingSetup()
+    {
+    	static const auto setup = MakeNoLogFileContext<CompressionRoundtripFuzzTestingSetup>();
+    	return setup.get();
+    }
+};
+
+void compression_roundtrip_initialize()
+{
+    SelectParams(ChainType::REGTEST);
+    rpc = InitializeCompressionRoundtripFuzzTestingSetup();
+    ctx = secp_context.GetContext();
+    FastRandomContext frandom_ctx(true);
```

Comment:
In c6f4427153b02f4b66535c9b8fdb0e9a2efa35cc:

If you are going to use a random context like this, it needs to be seeded from the fuzzer. (But I don't think you should use a random context. I'd just use the fuzzer directly to source all your data.)

Reply:
Fixed by using static seed to generate a bunch of data for the initialization step, Thanks!